project-lisa/source-code/client/src/api.ts

```diff
@@ -1,24 +1,261 @@
+import _git from "isomorphic-git"
+import _http from "isomorphic-git/http/web/index.js"
+import { createNodeishMemoryFs } from "@inlang-git/fs"
+import type { NodeishFilesystem } from "@inlang-git/fs"
+import { github } from "./github.js"
+import { wrap, transformRemote } from "./helpers.js"
+
 /**
- * Exports of the lisa client.
- */
-export type LisaClient = {
-	load: (
-		path: string,
-		options: {
-			fs: unknown
-			auth?: unknown
-		},
-	) => Promise<Repository>
-}
+ * Raw git cli api. (Used for legacy migrations)
+ *
+ * Used for legacy migrations. The idea of the `git-sdk` is to abstract working with git as
+ * a CLI. Sometimes access to the "raw" git CLI is required to
+ * achieve actions that are unsupported, or not yet, supported
+ * by the `git-sdk`. Esepcially useful for faster development
+ * iterations and progressively develop own api layer on top.
+*/
+export const raw = _git
+
+/**
+	* The http client for the git raw api. (Used for legacy migrations)
+	*
+	* Note: The http client is web-based. Node version 18 is
+	* required for the http client to work in node environments.
+*/
+export const http = _http
 
 export type Repository = {
-	fs: any
-	commit: () => unknown
-	push: () => unknown
-	pull: () => unknown
-	currentBranch: () => unknown
-	changeBranch: () => unknown
-	listRemotes: () => unknown
-	log: () => unknown
-	status: () => any
+	fs: NodeishFilesystem
+	commit: (args: { dir: string, author: any, message: string}) => Promise<Awaited<ReturnType<typeof raw.commit>> | undefined>
+	push: (args: { dir: string }) => Promise<Awaited<ReturnType<typeof raw.push>> | undefined>
+	pull: (args: { dir: string, author: any, fastForward: boolean, singleBranch: true }) => any
+	add: (args: { dir: string, filepath: string }) =>  Promise<Awaited<ReturnType<typeof raw.add>>>
+	listRemotes: () => Promise<Awaited<ReturnType<typeof raw.listRemotes>> | undefined>
+	log: (args: { dir: string, since: any}) =>  Promise<Awaited<ReturnType<typeof raw.log>>>
+	statusMatrix: (args: { dir:string, filter: any }) => Promise<Awaited<ReturnType<typeof raw.statusMatrix>>>
+	status: (args: { dir: string, filepath: string }) => Promise<Awaited<ReturnType<typeof raw.status>>>
```

Comment:
We can remove `dir` from every command. 

The `dir` is always (for now at least) the root of the repo.

Reply:
On that note, `load` should take a `workingDirectory` argument in the future once we support the opening of local repositories. 

Wait a sec, does `open` make more sense than `load` as entry point for repositories? 

- we don't "load" the repo as in "downloading" it once lazy fetching is implemented
- "opening" a local repository makes more sense than "loading" a local repository

```ts
const repository = open("my-repo.com", options)
```